src/foam/lib/PermissionPredicate.js

```diff
@@ -0,0 +1,55 @@
+/**
+@license
+Copyright 2020 The FOAM Authors. All Rights Reserved.
+http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+foam.CLASS({
+  package: 'foam.lib',
+  name: 'PermissionPredicate',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+  imports: [
+    'auth',
+  ],
+  properties: [
+    {
+      name: 'args',
+      class: 'StringArray'
+    }
+  ],
+  javaImports: [
+    'foam.core.X',
+    'foam.nanos.auth.AuthService'
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        X x = (X) obj;
+        AuthService auth = (AuthService) x.get("auth");
+        if ( args_ == null || args_.length <= 0 ) {
+          return true;
+        }
+
+        for ( String permission : args_ ) {
+          if ( auth.check(x, permission) )
+            return true;
+        }
+        return false;
+      `,
+      code: async function() {
```

Comment:
Does this work? Don't predicates need to return synchronously?

Reply:
so need to use slots, default to false, then update when the call returns.

Reply:
> Does this work? Don't predicates need to return synchronously?

I believe it does. I'm not sure where I can find information about that

Reply:
> so need to use slots, default to false, then update when the call returns.

But other predicates return boolean